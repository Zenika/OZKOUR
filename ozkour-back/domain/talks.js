const dayjs = require('dayjs')
const { getTalkFromDate } = require('../infrastructure/googlesheets/sheets.js')

const getTalk = async (start, end = dayjs()) => {
  const formatedDateStart = dayjs(start)
  const formatedDateEnd = dayjs(end)
  if (formatedDateStart.format('MM/YYYY') === formatedDateEnd.format('MM/YYYY')) {
    const param = { start: formatedDateStart.format('DD/MM/YYYY'), end: formatedDateEnd.format('DD/MM/YYYY') }
    const res = await getTalkFromDate(param)
    return res
  } else {
    let res = []
    let tempDateStart = formatedDateStart
    let tempDateEnd = formatedDateStart.add(1, 'month')
    while (tempDateStart.format('MM/YYYY') !== formatedDateEnd.format('MM/YYYY')) {
      const param = { start: tempDateStart.format('DD/MM/YYYY'), end: tempDateEnd.format('DD/MM/YYYY') }
      res = res.concat(await getTalkFromDate(param))
      tempDateStart = tempDateStart.add(1, 'month')
      tempDateEnd = tempDateStart.add(1, 'month')
    }
    const param = { start: tempDateStart.format('DD/MM/YYYY'), end: formatedDateEnd.format('DD/MM/YYYY') }
    res = res.concat(await getTalkFromDate(param))

    return res
  }
}

module.exports = {
  getTalk
}
