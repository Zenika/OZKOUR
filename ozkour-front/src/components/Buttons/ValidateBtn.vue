<script>
// import {ref} from "vue";
import axios from "axios";
// import { useTalkStore } from "../../stores/talks";

export default {

    data() {
        function sendTalks() {
            console.log("tyi");
            axios
            .post('http://localhost:3000/selected-talks', this.talk)
            .then((response) => console.log(response))
        }

        return {
            talk: {
                date: "",    
                universe: "",
                eventType: "",
                eventName: "",
                talkTitle: "",
                speakers: "",
            },
            sendTalks
        }
    },
};

</script>

<template>
    <button type="button" class="validate-btn" @click="sendTalks" >VALIDER</button>
</template>

<style scoped>
    .validate-btn{
        /* width: 15%; */
        background: linear-gradient(90deg, #EE2238 0%, #C21E65 100%);
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
        color: #FFFFFF;
        font-size: 18px;
        font-weight: 800;
        border: none;
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
        border-radius: 30px;
        padding: 10px 30px;
        cursor: pointer;
    }
    .validate-btn:hover {
        background: black;
        color: white;
    }
</style>