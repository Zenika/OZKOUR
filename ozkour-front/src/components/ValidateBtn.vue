<script >

</script>

<template>
    <button class="validate-btn">VALIDER</button>
</template>

<style scoped>
.validate-btn{
    width: 15%;
    background-color: #FFFFFF;
    color: #BF1D67;
    font-weight: 800;
    border: none;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    border-radius: 30px;
    padding: 15px;
    cursor: pointer;
}
</style>