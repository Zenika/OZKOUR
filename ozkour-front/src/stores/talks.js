// @ts-check
import dateFormat from "dateformat";
import { defineStore } from "pinia";

export const useTalkStore = defineStore({
  id: "talk",
  state: () => ({
    retrived: [],
    selected: [],
    blured: false,
    template: "",
    date: {}
  }),
  getters: {
    items: (state) =>
      state.retrived.reduce((items) => {
        return items;
      }, []),
  },
  actions: {
    updateTalks(newTalks) {
      this.retrived = newTalks;
    },
    updateCheckedTalks(newTalks) {
      this.selected = newTalks;
    },
    addCheckedTalk(newTalk) {
      let index = -1;
      let founded = false;
      while (index <= this.selected.length - 1 && !founded) {
        index++;
        // console.log("old " +this.selected[index]?.date)
        // console.log("new "+newTalk.date)
        // console.log("res " + this.selected[index]?.date > newTalk.date)
        if (this.selected[index]?.date > newTalk.date) founded = true;
      }
      this.selected.splice(index, 0, newTalk);
      //this.selected.push(newTalk);
    },
    removeCheckedTalk(oldTalk) {
      this.selected = this.selected.filter(function (value) {
        // console.log(value.talkTitle)
        // console.log(oldTalk.talkTitle)
        // console.log(value.talkTitle != oldTalk.talkTitle)
        return value.talkTitle != oldTalk.talkTitle;
      });
    },
    blur(){
      this.blured = true;
    },
    clarify(){
      this.blured = false;
    },
    // Visuel choisi
    pickedTemplate(chosenTemplate) {
      this.template = chosenTemplate
      console.log('template',chosenTemplate);
    },
    // Plage de date choisie
    selectedDate(start, end) {
      start = dateFormat(Date.parse(start.value), "dd/mm/yyyy");
      end = dateFormat(Date.parse(end.value), "dd/mm/yyyy");
      this.date = {start, end}
      console.log('start date',start);
      console.log('end date',end);
    }
  },
});


