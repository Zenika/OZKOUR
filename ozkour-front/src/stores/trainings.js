import qs from 'qs'
import { defineStore } from 'pinia'
import { api } from '@/api/apiConfig'

export const useTrainingStore = defineStore({
  id: 'training',
  state: () => ({
    retrieved: [],
    retreivingTrainings: false
  }),
  getters: {
    getSelectedTrainings: (state) =>
      state.retrieved.filter((training) => {
        return training.checked
      })
  },
  actions: {
    updateTrainings (newTrainings) {
      this.retrieved = newTrainings
    },
    checkTraining (selected) {
      this.retrieved.find(training =>
        training.trainingTitle === selected.trainingTitle).checked = true
    },
    uncheckTraining (selected) {
      this.retrieved.find(training =>
        training.trainingTitle === selected.trainingTitle).checked = false
    },
    async generateVisualForSelectedTrainings (templateName) {
      switch (templateName) {
      case 'E-mailing': {
        const { data } = await api
          .post('/training/emailing', this.getSelectedTrainings)
        return data.link
      }
      default:
        console.error('template :"', templateName, "\" n'est pas reconnu")
        return '/'
      }
    },
    async getTrainings (dateStart, dateEnd) {
      this.retreivingTrainings = true
      try {
        const { data } = await api
          .get('/training', {
            params: {
              start: dateStart,
              end: dateEnd
            },
            paramsSerializer: (params) => qs.stringify(params, { encode: false })
          })
        this.updateTrainings(data)
      } finally {
        this.retreivingTrainings = false
      }
    }
  }
})
