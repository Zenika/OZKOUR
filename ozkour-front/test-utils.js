export * from "@testing-library/vue";
export * from "@testing-library/user-event";
