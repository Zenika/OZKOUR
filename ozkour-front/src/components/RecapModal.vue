<script setup>
import ValidateBtn from './Buttons/ValidateBtn.vue'
import { useTalkStore } from '../stores/talks'
import axios from 'axios';
// import { dateStart, dateEnd } from '../components/ChoosingDate.vue'

const talk = useTalkStore()

const sendTalks = () => {
    console.log('talks', talk.getSelectedTalks);
    axios
    .post('http://localhost:3000/selected-talks', talk.getSelectedTalks)
    .then((response) => console.log('res :',response))
}
</script>

<template>
    <div class="popUp-bg">
        <div class="popUp-header">
            <button type="button" class="close-btn" @click="$emit('close')"> X </button>
            <h2>RÃ©capitulatif</h2>
        </div>

        <div class="recap">
            <div class="recap-details">
                <div class="icon-bg">
                    <img src="../assets/images/gallery.png" alt="calendar" class="icon">
                </div>
                <p data-test="template-detail"><b>Visuel : </b>{{ talk.template.template }}</p>
            </div>
            <div class="recap-details">
                <div class="icon-bg">
                    <img src="../assets/images/calendar.png" alt="calendar" class="icon">
                </div>
                <p data-test="date-detail"><b>Dates : </b>{{ talk.date.start }} au {{ talk.date.end }}</p>
            </div>
            <div>
                <div class="recap-details">
                    <div class="icon-bg">
                        <img src="../assets/images/microphone.png" alt="calendar" class="icon">
                    </div>
                    <p><b>Liste des talks : </b></p>
                </div>
                <ul class="events">
                    <li data-test="talk-title" v-for="talk in talk.getSelectedTalks" v-bind:key="talk">{{ talk.talkTitle }}</li>
                </ul>
            </div>
        </div>

        <div class="validate">
            <ValidateBtn @click="$emit('close'),sendTalks"/>
        </div>

    </div>
</template>

<style scoped>
    .popUp-bg {
        width: 35%;;
        top: 30%;
        bottom: auto;
        background: #F2F2F2;
        position: absolute;
        z-index: 20;
        transform: 'translate(-50%, -50%)';
        border-radius: 20px;
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    }

    .popUp-header{
        display: flex;
        justify-content: center;
        align-items: center;
        background: linear-gradient(90deg, #EE2238 0%, #C21E65 100%);
        border-radius: 20px 20px 0px 0px;
    }

    .close-btn {
        position: absolute;
        left: 5%;
        align-self: flex-start;
        border: none;
        color: white;
        font-family: 'Open Sans', sans-serif;
        font-size: 20px;
        font-weight: bold;
        cursor: pointer;
        padding-top: 10px;
        background: transparent;
    }

    .recap{
        /* width: 100%; */
        color: black;
        padding: 0px 20px;
    }

    .icon-bg {
        width: 30px;
        height: 30px;
        display: flex;
        justify-content: center;
        align-items: center;
        border-radius: 50%;
        background: linear-gradient(90deg, #EE2238 0%, #C21E65 100%);
        padding: 2px;
    }

    .icon {
        width: 20px;
    }

    .recap-details {
        display: flex;
        flex-direction: row;
        align-items: center;
        gap: 10px;
    }

    .events {
        margin: 0px;
        padding-left: 60px;
    }

    .validate {
        display: flex;
        justify-content: center;
        padding: 30px 0px;
    }

</style>