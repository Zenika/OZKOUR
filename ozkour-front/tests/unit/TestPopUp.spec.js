import { shallowMount } from "@vue/test-utils";
import { createTestingPinia } from "@pinia/testing";
import RecapModal from "@/components/RecapModal.vue"
import ValidateBtn from '@/components/Buttons/ValidateBtn.vue'
import { useTalkStore } from "../../src/stores/talks";

// On utilise 'shallowMount' pour ne pas mount les composants enfants (ici ValidateBtn)

describe("RecapModal Component", () => {
    beforeEach(() => {

    })
    //--- Afficher le visuel choisi par l'utilisateur
    it("Display the template chosen", async () => {
        const wrapper = shallowMount(RecapModal, {
            global: {
              plugins: [
                createTestingPinia({
                  initialState: {
                    talk: { 
                        retrieved: talks,
                        template : {
                            template : 'E-mailing', 
                            frequency : 'month' 
                        }
                    },
                  },
                }),
              ],
            },
        });

        const template = wrapper.find('[data-test="template-detail"]').text()
        
        expect(template).toBe('Visuel : E-mailing')
    })

    //--- Afficher la plage de dates choisi par l'utilisateur
    it("Display the date range chosen", async () => {
        const wrapper = shallowMount(RecapModal, {
            global: {
              plugins: [
                createTestingPinia({
                  initialState: {
                    talk: { 
                        retrieved: talks,
                        date : { 
                            start : '11/01/2021', 
                            end : '12/02/2021'
                        }
                    },
                  },
                }),
              ],
            },
        });

        const dateText = wrapper.find('[data-test="date-detail"]').text()

        expect(dateText).toBe('Dates : 11/01/2021 au 12/02/2021')

        // console.log(dateText.html()); => affiche la partie HTML du code
    }),

    //--- Afficher la liste des titres des talks choisi par l'utilisateur 
    it("Display a list of the chosen talks, only their titles", async () => {
        const wrapper = shallowMount(RecapModal, {
            global: {
              plugins: [
                createTestingPinia({
                    plugins: [useTalkStore],
                    initialState: {
                        talk: { retrieved: talks },
                    },
                }),
              ],
            },
        });
    
        const talkTitle = wrapper.find('.events').html()

        expect(talkTitle).toMatchSnapshot()
    })

    //--- Bouton fermer la pop up
    it("Close the pop up when you click on the cross button", async () => {
        const wrapper = shallowMount (RecapModal)

        wrapper.find('.close-btn').trigger('click')

        expect(wrapper.emitted('close')).toHaveLength(1)
    })
})

const talks =
  [
    {
      date: "19/01/2021",
      universe: "",
      eventType: "Meetup",
      eventName: "GraalVM Night",
      talkTitle: "GraalVM for Sustainable Software Development?",
      speakers: "Adrien Nortain",
      checked: true,
    },
    {
      date: "19/01/2021",
      universe: "",
      eventType: "NightClazz",
      eventName: "NightClass",
      talkTitle: "Migration JS vers TS sur du react",
      speakers: "Jules Hablot",
      checked: true,
    },
    {
      date: "21/01/2021",
      universe: "",
      eventType: "Meetup",
      eventName: "Nantes JS #55",
      talkTitle: "Nuxt 2021",
      speakers: "Yann Bertrand",
      checked: true
    },
    {
      date: "21/01/2021",
      universe: "",
      eventType: "Autre",
      eventName: "Webinar Strigo",
      talkTitle: "Simplify Remote Hands-On Training and Improve Engagement",
      speakers: "Yoan Rousseau / Oliver Huber",
      checked: true
    },
    {
      date: "25/01/2021",
      universe: "",
      eventType: "NightClazz",
      eventName: "RemoteClazz Nodejs",
      talkTitle: "Techniques minimalistes pour Node.js",
      speakers: "Hugo Wood",
      checked: false,
    }
];
