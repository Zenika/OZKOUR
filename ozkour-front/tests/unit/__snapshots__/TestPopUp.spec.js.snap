// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecapModal Component Display a list of the chosen talks, only their titles 1`] = `
<ul class="events">
  <li data-test="talk-title">GraalVM for Sustainable Software Development?</li>
  <li data-test="talk-title">Migration JS vers TS sur du react</li>
  <li data-test="talk-title">Nuxt 2021</li>
  <li data-test="talk-title">Simplify Remote Hands-On Training and Improve Engagement</li>
</ul>
`;
